.rubocop_todo.yml

```diff
@@ -134,6 +134,8 @@ Metrics/ModuleLength:
 # Configuration parameters: CountKeywordArgs, MaxOptionalParameters.
 Metrics/ParameterLists:
   Max: 7
+  Exclude:
+    - 'lib/cucumber/core/test/case.rb'
```

Comment:
First pass of brief review. If you're able to tidy these up in the next 1-2 weeks feel free to do so, as this won't be merged for a couple of weeks.

Reply:
My suggested `C::C::Test::Case` kwargs refactor would address this... unless there's a similar cop for max kwargs??

Just checked, and its a bit of a grey area. By default, it would indeed complain about the kwargs list being too long (default is 3, whether its posargs or kwargs). However, we can set the `CountKeywordArgs` flag to `false` to not count the latter.

Interestingly, there's a hard-coded exception carved out for constructors for `Struct` and `Data`, and I quote (emphasis mine): 
> Any number of arguments for initialize method inside a block of Struct.new and Data.define is always allowed. **This is because checking the number of arguments of the initialize method does not make sense.**

https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Metrics/ParameterLists

So, I would assert that by RuboCop's own logic, this cop already doesn't make sense for our use-case either, so we should keep it as an explicit exception either way!

Reply:
@luke-hill More thoughts here.

Ideally, we'd refactor `C::C::Test::Case` to inherit from `Data`, because it seems to me that `Data` perfectly fits its use-case and intent, as @mattwynne mentioned in another comment farther down. But `Data` is Ruby 3.2+ only. We could polyfill with https://github.com/saturnflyer/polyfill-data, but that's a new dependency... so maybe we could copy and paste it in-repo with some minor tweaks:

https://github.com/saturnflyer/polyfill-data/blob/main/lib/data.rb

Reply:
If we think something in v3.2 will enable us to refactor this better I'm happy waiting until then. There's 1000 other things I'm trying to refactor.